src/system.rs

```diff
@@ -284,6 +284,11 @@ pub trait SystemData<'a> {
     ///
     /// Please note that returning wrong dependencies can lead to a panic.
     fn writes() -> Vec<ResourceId>;
+
+    /// Returns names of retrieved resources.
+    fn resource_names() -> Vec<&'static str> {
+        vec![std::any::type_name::<Self>()]
+    }
```

Comment:
I just noticed that this PR is adding a new method to the SystemData trait. What is the purpose here? Was it intended to be included in this PR?

Reply:
Oh huh, I hadn't even noticed that. Must be from some testing a long time ago